src/Controllers/UserController.php

```diff
@@ -22,17 +26,31 @@ public function setService(AbstractService $userService)
 
     public function listAll(RequestInterface $request)
     {
-        // TODO create a check function for session
-        if ($this->session->get('role') === 'admin') {
-            try {
-                $entities = $this->userService->getAll($request);
-                return $this->renderer->renderView("admin-users-listing.phtml", ['session' => $this->session, 'entities' => $entities]);
-            } catch (NoSuchRowException $e) {
-                return $this->renderer->renderView("admin-users-listing.phtml", ['session' => $this->session]);
-            }
+        if ($this->session->get('role') !== 'admin') {
+            return $this->renderer->renderView("login.phtml", []);
         }
 
-        return $this->renderer->renderView("login.phtml", []);
+        $page = $request->getParameter('page');
+        $paginator = $this->createPaginator((int)$page);
+        try {
+            $entities = $this->userService->getAll($paginator->getCurrentPage());
+            return $this->renderer->renderView(
```

Comment:
this return should not be in the `try` block.

only the `userService->getAll` method throws the expected exception